servo-atlas/src/main/java/com/netflix/servo/publish/atlas/AtlasMetricObserver.java

```diff
@@ -306,8 +324,25 @@ protected int getNumberOfCopies() {
     return 1;
   }
 
+
+  private String getPayloadFilename() {
+    SimpleDateFormat fmt = new SimpleDateFormat(FILE_DATE_FORMAT);
+    fmt.setTimeZone(TimeZone.getTimeZone("UTC"));
+    return fmt.format(new Date());
+  }
+
   protected Observable<Integer> getSenderObservable(TagList tags, Metric[] batch) {
     JsonPayload payload = new UpdateRequest(tags, batch, batch.length);
+    if (shouldDumpPayload()) {
+      try {
+        String name = getPayloadFilename();
+        File out = new File(getPayloadDirectory(), name);
+        JsonGenerator generator = factory.createGenerator(out, JsonEncoding.UTF8);
+        payload.toJson(generator);
```

Comment:
Not sure if jackson supports it, but if there is an easy way to format with one metric per line, then it would be easier to grep. We can use jq, but it requires a bit more knowledge of the structure to get started.

Reply:
I've updated the PR to make it easier to grep. One metric per line.